doc/versioning.md

```diff
@@ -0,0 +1,32 @@
+# gRPC-dotnet package versioning policies
+
+## Overview
+
+This document covers versioning policies of the packages produced in this repo:
+
+- Grpc.AspNetCore
+- Grpc.AspNetCore.Server
+- Grpc.AspNetCore.Server.ClientFactory
+- Grpc.Net.Client
+- Grpc.Net.ClientFactory
+- Grpc.Net.Common
+- dotnet-grpc
+
+## Design considerations
+
+### Major version number
+
+The initial major version number was set to 2 to match the release of corresponding Grpc.Core packages. Going forward, the major version will be incremented when there's a need to ingest a major Grpc.Core or ASP.NET Core dependency for breaking changes.
```

Comment:
We still need to comply with https://github.com/grpc/grpc/blob/master/doc/versioning.md. It would be good to use wording that makes it clear we won't bump major version unless necessary.